packages/react/src/utils/useWaitForComponentMount.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Used to detect when a Clerk component has been added to the DOM.
+ */
+function waitForElementChildren(options: { selector?: string; root?: HTMLElement | null; timeout?: number }) {
+  const { root = document?.body, selector, timeout = 0 } = options;
+
+  return new Promise<void>((resolve, reject) => {
+    if (!root) {
+      reject(new Error('No root element provided'));
+      return;
+    }
+
+    let elementToWatch: HTMLElement | null = root;
+    if (selector) {
+      elementToWatch = root?.querySelector(selector);
+    }
+
+    // Check if the element already has child nodes
+    const isElementAlreadyPresent = elementToWatch?.childElementCount && elementToWatch.childElementCount > 0;
+    if (isElementAlreadyPresent) {
+      resolve();
+      return;
+    }
+
+    // Set up a MutationObserver to detect when the element has children
+    const observer = new MutationObserver(mutationsList => {
+      for (const mutation of mutationsList) {
+        if (mutation.type === 'childList') {
+          if (!elementToWatch && selector) {
+            elementToWatch = root?.querySelector(selector);
+          }
```

Comment:
There's most definitely a smarter way to do this, it's not the most efficient to re-query on each observed mutation, but given that it's scoped to the time the component mounts to the time the clerk-js mount happens it shouldn't be running too many times...